client/kv.go

```diff
@@ -78,12 +70,28 @@ type KV struct {
 // create a transactional client, the KV struct should be manually
 // initialized in order to utilize a txnSender. Clock is used to
 // formulate client command IDs, which provide idempotency on API
-// calls. If clock is nil, uses time.UnixNanos as default
+// calls and defaults to the system clock.
 // implementation.
-func NewKV(sender KVSender, clock Clock) *KV {
+func NewKV(ctx *Context, sender KVSender) *KV {
+	if ctx == nil {
+		ctx = NewContext()
+	}
 	return &KV{
-		sender: sender,
-		clock:  clock,
+		sender:          sender,
+		User:            ctx.User,
```

Comment:
No, it will be fine. The way we influence the configuration from some of the tests needs refactoring. I think we also have some cases in which the test is very far away from internal client objects. These internal clients should have their own configuration somewhere instead of using the default (but that's another issue).